addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/api/UniFiClient.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.unifi.internal.api;
+
+import org.apache.commons.lang.BooleanUtils;
+import org.openhab.binding.unifi.internal.api.json.UniFiMacDeserializer;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+/**
+ * The {@link UniFiClient} is the base data model for any (wired or wireless) connected to a UniFi network.
+ *
+ * @author Matthew Bowman - Initial contribution
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "is_wired", defaultImpl = UniFiUnknownClient.class)
+@JsonSubTypes({ @JsonSubTypes.Type(value = UniFiWiredClient.class, name = "true"),
+        @JsonSubTypes.Type(value = UniFiWirelessClient.class, name = "false") })
+public abstract class UniFiClient {
+
+    @JsonProperty("_id")
+    protected String id;
+
+    @JsonDeserialize(using = UniFiMacDeserializer.class)
+    protected String mac;
+
+    protected String hostname;
+
+    @JsonProperty("site_id")
```

Comment:
You can get rid of this problem in one go:

https://stackoverflow.com/a/11810183/3656774